src/server/mod.rs

```diff
@@ -18,12 +21,42 @@ pub use encode::Encoder;
 pub struct ServerOptions {
     /// Timeout to handle headers. Defaults to 60s.
     headers_timeout: Option<Duration>,
+    default_host: Option<String>,
+}
+
+impl ServerOptions {
+    /// constructs a new ServerOptions with default settings
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// sets the timeout by which the headers must have been received
+    pub fn with_headers_timeout(mut self, headers_timeout: Duration) -> Self {
+        self.headers_timeout = Some(headers_timeout);
+        self
+    }
+
+    /// Sets the default http 1.0 host for this server. If no host
+    /// header is provided on an http/1.0 request, this host will be
+    /// used to construct the Request Url.
+    ///
+    /// If this is not provided, the server will respond to all
+    /// http/1.0 requests with status `505 http version not
+    /// supported`, whether or not a host header is provided.
+    ///
+    /// The default value for this is None, and as a result async-h1
+    /// is by default an http-1.1-only server.
+    pub fn with_default_host(mut self, default_host: &str) -> Self {
+        self.default_host = Some(default_host.into());
+        self
+    }
```

Comment:
Side note: I'm increasingly convinced that async-h1 could be nicer to use if we had concrete `Client` and `Server` structs, and not the intermediate short-hand constructors. Rather than using a separate `Options` struct, exposing the `host` param through the constructor may be easier:

```rust
let mut server = Server::new();
assert_eq!(server.host(), None);

let mut server = Server::with_host("website.com");
assert_eq!(server.host(), Some("website.com"));
```

Reply:
Entirely agreed

Reply:
That's been my intent for a few PRs — I've been leaving the functions around until the interface for the structs feels right. My hunch is that implementing future for the struct will simplify a bunch of things